### DISCUSSION TITLE
Post man ERROR 404

### DISCUSSION THREAD

**Discussion by Gdorogoy** [community/community#68535] [Programming Help] [answered]
### Select Topic Area

Question

### Body

**i have written a java [(link:](https://github.com/Gdorogoy/GymBuddy/tree/main/Gym-Buddy))on spring, i use mongodb as database but when i try to post some data on postman it gives error 404 not found.**
i only watched some youtube tutorial on spring and have not much knowldge.
PLS HELP🙏🙏

**also it gives this in the run terminal in intelij:**
_2023-09-28T18:27:20.239+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : POST "/api/add-exercise", parameters={}
2023-09-28T18:27:20.250+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2023-09-28T18:27:20.261+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2023-09-28T18:27:20.262+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Completed 404 NOT_FOUND
2023-09-28T18:27:20.272+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : "ERROR" dispatch for POST "/error", parameters={}
2023-09-28T18:27:20.274+03:00 DEBUG 27403 --- [nio-8080-exec-1] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController#error(HttpServletRequest)
2023-09-28T18:27:20.335+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2023-09-28T18:27:20.337+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{timestamp=Thu Sep 28 18:27:20 IDT 2023, status=404, error=Not Found, message=No message available,  (truncated)...]
2023-09-28T18:27:20.451+03:00 DEBUG 27403 --- [nio-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Exiting from "ERROR" dispatch, status 404_

**Comment by pvfarooq** [comment, marked as answer]
Hi @Gdorogoy,  I'm not familiar with Spring Boot, but based on the error message, it seems like there might be an issue with your code. The error suggests that there is no _'/api/add-exercise'_ endpoint in your controller. To fix this, ensure that you didn't accidentally put a '/' at the beginning of the endpoint in your controller code.

Additionally, make sure that your controller is properly registered in the top-level controller configuration. This means that Spring Boot should know about your controller so it can route requests to the correct endpoints.

In simpler terms, double-check your code to make sure there are no mistakes in the endpoint definition, and ensure that your controller is set up correctly so that Spring Boot knows how to handle requests to that endpoint.

**Comment by MateoRodriguez0** [comment]
I suppose you could solve it, but just in case the error is in postman, you are making the request with a POST method and in your project you do not have that url mapped to the post method
Here is the error
POST "/api/add-exercise", parameters ={}
the method does not correspond to the url
